de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/MetadataTemplateViewDialog.java

```diff
@@ -100,45 +111,63 @@ public MetadataTemplateViewDialog(MetadataView.Presenter presenter,List<Avu> tem
 
                 avu.setValue(value);
             }
-
             avus.add(avu);
-        }
+
+        });
         return avus;
     }
-    
+
     public boolean isValid() {
-      List<IsField<?>> fields = FormPanelHelper.getFields(widget);
-            for (IsField<?> f : fields) {
-                if (!f.isValid(false)) {
-                    valid = false;
-                }
+        List<IsField<?>> fields = FormPanelHelper.getFields(widget);
+        for (IsField<?> f : fields) {
+            if (!f.isValid(false)) {
+                valid = false;
             }
+        }
 
-      return valid;
+        return valid;
     }
-    
+
     private void buildAvuMap() {
-    	for (MetadataTemplateAttribute attribute: attributes) {
-    		Avu md = findMetadataForAttribute(attribute.getName());
-    		templateAttrAvuMap.put(attribute.getName(),md);
-    	}
+        for (MetadataTemplateAttribute attribute : attributes) {
+            List<Avu> mds = findMetadataForAttribute(attribute.getName());
+            if (mds != null && !mds.isEmpty()) {
+                mds.forEach(md -> {
+                    String tag = getTagForMetadata(md);
+                    templateTagAvuMap.put(tag, md);
+                    templateTagAtrrMap.put(tag, attribute);
+                    // GWT.log(tag + "-->" + attribute.getName());
+                });
+            } else {
+                Avu avu = presenter.setAvuModelKey(MetadataPresenterImpl.newMetadata(attribute.getName(),
+                                                                                     "",
+                                                                                     "")); //$NON-NLS-1$ //$NON-NLS-2$
+                String tag = getTagForMetadata(avu);
+                templateTagAvuMap.put(tag, avu);
+                templateTagAtrrMap.put(tag, attribute);
+                // GWT.log(tag + "-->" + attribute.getName());
+            }
+
+        }
     }
-    
-    private Avu findMetadataForAttribute(String attribute) {
-    	if(templateMd != null) {
-	    	for(Avu md: templateMd) {
-	    		if(md.getAttribute().equals(attribute)) {
-	    			return md;
-	    		}
-	    	}
-    	}
-		return null;
+
+    private List<Avu> findMetadataForAttribute(String attribute) {
```

Comment:
These are both good points. One thing to keep in mind when trying to decide if something should go into a view vs. a presenter is that if we switch the view to something else, ideally we wouldn't have to change anything in the main Metadata presenter. So in that respect, these FastMaps may belong to the view, since if we port the view to something like React, then maybe FastMaps and ListStores would no longer be required.

On the other hand, it may make porting the view to something else easier if the view was simple and dumb. So in that case, maybe it would help if this view had its own dialog-preseneter for manipulating the FastMaps and creating new AVUs, then as @ash3rz said, the view would only have to be concerned with displaying the models its given and calling events back to its presenter (which could in turn coordinate with the main presenter). Then if this view is ported to something else, its dialog-preseneter may or may not still be required.

These are just some general design ideas I wanted to add to this discussion, and not specific to the changes for this PR.